cylc/flow/scheduler.py

```diff
@@ -1262,11 +1262,17 @@ def _configure_contact(self) -> None:
 
     def load_flow_file(self, is_reload=False):
         """Load, and log the workflow definition."""
-        # Local workflow environment set therein.
-        # Allow -S and -D to take effect in Cylc VR.
-        # https://github.com/cylc/cylc-flow/issues/5968
-        self.options.rose_template_vars = []
-        self.options.defines = []
+        if is_reload:
+            # If the workflow is reloaded clear any existing rose options
+            # The reload command doesn't have the ability to set these but
+            # if the user has used VR to re-install before reload they will
+            # expect the changed values not the ones stored on the scheduler.
+            # Note: Does NOT apply to reload on restart, because the play
+            # command used to restart can have template variables attached.
```

Comment:
I'm a bit confused
- If the user has used `cylc vr running-workflow -S` (reinstall + reload), how are the changed values getting through to the scheduler then?
- What do you mean by "reload on restart"? If the user has used `cylc vr stopped-workflow -S` (reinstall + restart), then there is no reload, right?

Reply:
> If the user has used cylc vr running-workflow -S (reinstall + reload), how are the changed values getting through to the scheduler then?

The reinstall updates the `rose-suite-cylc-install.conf` which the reload reads. The reload doesn't have the `-S` option so clearing that here only affects options set on the scheduler by an earlier Cylc play.

Your question has however revealed a horrid case where this doesn't work:

```
cylc vip -S 'VAR=1' --pause
cylc vr bugs/5968
```

Do you think that the correct options might be to allow Cylc Reload to set `-S` and it's friends, becuase to me that looks like the simplest way of doing things?


> What do you mean by "reload on restart"? If the user has used cylc vr stopped-workflow -S (reinstall + restart), then there is no reload, right?

Oliver described it as an implicit reload.

Reply:
> Your question has however revealed a horrid case where this doesn't work

What does happen with this example, and what is meant to happen?